pyEpiabm/pyEpiabm/routine/spatial_infection.py

```diff
@@ -0,0 +1,88 @@
+#
+# Calculate spatial force of infection based on Covidsim code
+#
+
+from pyEpiabm.core import Person, Cell, Parameters
+
+
+class SpatialInfection:
+    """Class to calculate the infectiousness and susceptibility
+    parameters for the force of infection parameter, between cells.
+    """
+    @staticmethod
+    def cell_inf(inf_cell: Cell, timestep: int):
+        """Calculate the infectiousness of one cell
+        towards its neighbouring cells.
+
+        :param inf_cell: Cell doing infecting
+        :type inf_cell: Cell
+        :param timestep: Current simulation timestep
+        :type timestep: int
+        :return: Average number of infection events from the cell
+        :rtype: int
+        """
+        R_0 = Parameters.instance().basic_reproduction_num
+        total_infectors = inf_cell.number_infectious()
+        # Add in other classes of people who are infectors
+        number_to_infect = total_infectors * R_0
+        # Round this float to nearest integer
+        return round(number_to_infect)
```

Comment:
I'm not certain this is the best behaviour - say this is 1.5 then all cells would have value 2. Surely we would want half of cells to have value 1 and half to have value 2 (randomly assigned)? Do we know how CovidSim does it?

Reply:
Implemented with choice weighted by how close the float is to respective neighbouring integers